api/src/db-connection.ts

```diff
@@ -389,6 +409,92 @@ export class Connection {
     return entities;
   }
 
+  /**
+   * Default implementation in `getEntitiesForPaper` naively retrieves all entities,
+   * which includes quadratic data duplication across multiple instances of the same symbol.
+   * This variant method is meant as a placeholder bridge until the underlying DB schema and
+   * extraction write layer is changed to be non-duplicative.
+   * Leaving the underlying data unchanged in the DB, we selectively exclude the entity data
+   * types known to be redundant. These are retrieved later and added into lookup tables.
+   */
+  async getDedupedEntitiesForPaper(paperSelector: PaperSelector, entityTypes: EntityType[], version?: number) {
+    const entities = await this.getEntitiesForPaper(paperSelector, entityTypes, false, version);
+
+    // Short-circuit fancy behavior below if we don't need the shared symbol data.
+    if (entityTypes.indexOf("symbol") === -1 && entityTypes.length > 0) {
+      return {
+        entities
+      }
+    }
+
+    // To provide a deduped copy of shared data between symbol instances, we
+    // identify each symbol by its "disambiguated" id (currently the `mathml` attribute).
+    // An arbitrary symbol entity from within each `mathml` is chosen as an exemplar from
+    // which to look up supporting data that is identical across instances.
+    const disambiguatedSymbolIdsToExemplarEntityIds = entities
+      .filter((row) => isSymbol(row))
+      .reduce(
+        (dict, symbol) => {
+          const disambiguatedId = (symbol as Symbol).attributes.mathml;
```

Comment:
```suggestion
        (dict, symbol: Symbol) => {
          const disambiguatedId = symbol.attributes.mathml;
```
This might also work.

Reply:
Sadly it does not. The explicit cast in 438 appears necessary, even though symbol is recognized as a `Symbol` in the signature.